src/components/wizards/create-vm/CreateVMWizard.js

```diff
@@ -0,0 +1,366 @@
+import React from 'react';
+import { Form, FormControl, Col, ButtonGroup, DropdownButton, MenuItem, Checkbox, FormGroup, ControlLabel, HelpBlock, Wizard } from 'patternfly-react';
+import './CreateVMWizard.css';
+import * as _ from 'lodash';
+import { StorageTable } from './StorageTable';
+
+const defaultState = {
+  activeStepIndex: 0,
+  basicSettings: {},
+  wizardValid: false
+}
+
+export class CreateVMWizard extends React.Component {
+
+    constructor(props){
+      super(props);
+      this.state = defaultState;
+    }
+  
+    onFormChange = (newValue, target) => {
+      let valid = null;
+      if (this.basicSettings[target].required && newValue.trim().length === 0){
+        valid= `${this.basicSettings[target].title} is required`;
+      }
+      const basicSettings = {
+        ...this.state.basicSettings,
+        [target]: {
+          value: newValue,
+          valid: valid
+        }
+      };
+      this.setState({
+        basicSettings: basicSettings
+      });
+      this.validateWizard(basicSettings);
+    }
+  
+    validateWizard = values => {
+      let wizardValid = true;
+
+      //check if all required fields are defined
+      const requiredKeys = Object.keys(this.basicSettings).filter(key => this.isFieldRequired(key, values));
+      const requiredKeysInValues = Object.keys(values).filter(key => this.isFieldRequired(key, values));
+      if(requiredKeys.length !== requiredKeysInValues.length){
+        wizardValid = false;
+      }
+
+      //check if all fields are valid
+      for (let key in values) {
+        if (values[key].valid && (this.basicSettings[key].isVisible ? this.basicSettings[key].isVisible(values) : true)) {
```

Comment:
The second part of the `if` condition can be written as
```
(this.basicSettings[key].isVisible && this.basicSettings[key].isVisible(values)) || true
```
instead of using the ternary operator. Even better, it could be extracted into a `const` variable so that the condition itself is easier to reason about.